plugin.audio.ampache/default.py

```diff
@@ -0,0 +1,1070 @@
+        addDir(ut.tString(30116) + " (" + ampache.getSetting("albums") + ")",None,2)
+        addDir(ut.tString(30118) + " (" + ampache.getSetting("playlists")+ ")",None,13)
+        apiVersion = int(ampache.getSetting("api-version"))
+        if apiVersion >= 380001:
+            addDir(ut.tString(30119),None,18)
+
+    #quick access
+    elif mode==25:
+        addDir(ut.tString(30135),9999994,2)
+        if(int(ampache.getSetting("api-version"))) >= 400001:
+            addDir(ut.tString(30162),9999989,2)
+            addDir(ut.tString(30153),9999992,2)
+            addDir(ut.tString(30191),9999988,2)
+        else:
+            #use recently added for old api versions
+            addDir(ut.tString(30127),9999997,6)
+        addDir(ut.tString(30147),9999994,3)
+
+    #highest
+    elif mode==30:
+        addDir(ut.tString(30149),9999993,1)
+        addDir(ut.tString(30150),9999993,2)
+        addDir(ut.tString(30151),9999993,3)
+
+    #frequent
+    elif mode==31:
+        addDir(ut.tString(30152),9999992,1)
+        addDir(ut.tString(30153),9999992,2)
+        addDir(ut.tString(30154),9999992,3)
+    
+    #flagged
+    elif mode==32:
+        addDir(ut.tString(30155),9999991,1)
+        addDir(ut.tString(30156),9999991,2)
+        addDir(ut.tString(30157),9999991,3)
+
+    #forgotten
+    elif mode==33:
+        addDir(ut.tString(30158),9999990,1)
+        addDir(ut.tString(30159),9999990,2)
+        addDir(ut.tString(30160),9999990,3)
+    
+    #newest
+    elif mode==34:
+        addDir(ut.tString(30161),9999989,1)
+        addDir(ut.tString(30162),9999989,2)
+        addDir(ut.tString(30163),9999989,3)
+    
+    #recent
+    elif mode==35:
+        addDir(ut.tString(30190),9999988,1)
+        addDir(ut.tString(30191),9999988,2)
+        addDir(ut.tString(30192),9999988,3)
+
+    #play track mode  ( mode set in add_links function )
+    #mode 45 to avoid endDirectory
+    elif mode==45:
+        #workaround busydialog bug
+        xbmc.executebuiltin('Dialog.Close(busydialog)')
+        play_track(object_id, m_params['song_url'])
+
+    #change rating
+    elif mode==47:
+        setRating()
+
+    #settings
+    elif mode==50:
+        ampache.openSettings()
+
+    #the four modes below are used to manage servers
+    elif mode==51:
+        if servers_manager.addServer():
+            servers_manager.switchServer()
+    
+    elif mode==52:
+        if servers_manager.deleteServer():
+            servers_manager.switchServer()
+    
+    elif mode==53:
+        servers_manager.modifyServer()
+    
+    elif mode==54:
+        servers_manager.switchServer()
+
+    #no end directory item ( problem with failed searches )
+    #endDir is the result of the search function
+    if endDir == False:
+        mode = modeMax
+
+    if mode == None or mode < endDirectoryMode:
+        xbmc.log("AmpachePlugin::endOfDirectory " + str(handle),  xbmc.LOGDEBUG)
+        xbmcplugin.endOfDirectory(handle)
+
+
+if (__name__ == '__main__'):
```

Comment:
usually it is a good practice to keep the entrypoint file (the one mentioned in addon.xml) as smaller as possible as it is the only file that is not compiled to bytecode:

```python
from myplugin import Main
Main()
```

Reply:
This is a big change, i have to close this push, test it and release a new version

Reply:
done